### ISSUE TITLE
vs2015 example compile error

### ISSUE THREAD

**Issue by darkr4y** [kenba/via-httplib#4]
Example file : example_https_server.cpp

Error C4996	'boost::asio::basic_socket<Protocol,StreamSocketService>::cancel': By default, this function always fails with operation_not_supported when used on Windows XP, Windows Server 2003, or earlier. Consult documentation for details.	Test_via-http	c:\users\johndoe\documents\via-httplib\include\via\comms\ssl\ssl_tcp_adaptor.hpp	188

**Comment by kenba** [comment]
Do you need to support Windows XP, Windows Server 2003, or earlier?

In `via-httplib.pro` the minimum version is set to `Windows 7` by the following macro:

    _WIN32_WINNT=_WIN32_WINNT_WIN7

Does that work for you?

**Comment by darkr4y** [comment]
Thx to reply ..
The Lib can only support NT6+?
I need to run something which using this lib on NT5 , is there a way to work out?

**Comment by kenba** [comment]
Error C4996 seems to be a relatively new "feature" of Visual Studio, see: [Why does visual studio 2013 error on c4996?](http://stackoverflow.com/questions/20448102/why-does-visual-studio-2013-error-on-c4996)

`via-httplib` is based on the `boost-asio` library which must be calling a deprecated function in `close`. According to [Microsoft](https://msdn.microsoft.com/en-us/library/ttcz0bys.aspx) you can disable the test by defining the macro:

    _CRT_NONSTDC_NO_WARNINGS

I would also recommend changing the macro `_WIN32_WINNT` to `_WIN32_WINNT` or `0x0501`
That should enable you to build for Windows XP.

**Comment by darkr4y** [comment]
Thx I will try and give a reply..

**Comment by darkr4y** [comment]
I change the via-httplib.pro file as you recommend and rebuild the via-http lib.
I also disable the SDL check , it will turn the error to warning.
but there is another error:
Error	C2064	Test_via-http	c:\boost_1_60_0\boost\asio\basic_io_object.hpp	Line 45

**Comment by darkr4y** [comment]
so its very kind of giving me a exmaple to compile the demo and test it ..

**Comment by darkr4y** [comment]
Finally I have compiled the demo.
Its not must to change the via-httplib.pro file
Just use the vs2015 **UPDATE1** , cuz UPDATE2 will hv a bug with using ASIO lib.
and the cmake should be 
```
cmake.exe -G "Visual Studio 14 2015" -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DVIA_HTTPLIB_BUILD_SHARED_LIBS=OFF ../
```
Thx for this lib & help me to work out the problem! :) Its time to close this issue.

**Comment by kenba** [comment]
Thank you for investigating this issue and documenting your work around.

Since it's unlikely that [boost issue 12115](https://svn.boost.org/trac/boost/ticket/12115) will be fixed before [boost release 1.62.0](http://www.boost.org/development/index.html), I've added a comment to the project `README.md` file, referencing this issue and I'm closing it as you request.
